x-pack/plugins/security_solution/public/detections/containers/detection_engine/rules/use_rule_from_timeline.tsx

```diff
@@ -103,18 +90,32 @@ export const useRuleFromTimeline = (setRuleQuery: SetRuleQuery): RuleFromTimelin
   );
   // end browser field management
 
+  const getInitialUrlParamValue = useGetInitialUrlParamValue<string>(RULE_FROM_TIMELINE_URL_PARAM);
+  const timelineIdFromUrl = useMemo(getInitialUrlParamValue, [getInitialUrlParamValue]);
+
+  const getInitialUrlParamValueEql = useGetInitialUrlParamValue<string>(RULE_FROM_EQL_URL_PARAM);
+  const timelineIdFromUrlEql = useMemo(getInitialUrlParamValueEql, [getInitialUrlParamValueEql]);
+
   // start set rule
   const handleSetRuleFromTimeline = useCallback(() => {
     if (selectedTimeline == null || selectedDataViewBrowserFields == null) return;
 
-    const newQuery = {
-      query: selectedTimeline.kqlQuery.filterQuery?.kuery?.expression ?? '',
-      language: selectedTimeline.kqlQuery.filterQuery?.kuery?.kind ?? 'kuery',
-    };
-    const newFilters = selectedTimeline.filters ?? [];
+    const isEql = timelineIdFromUrlEql != null;
+    const newQuery = isEql
+      ? {
+          query: selectedTimeline.eqlOptions.query ?? '',
+          language: 'eql',
+        }
+      : {
+          query: selectedTimeline.kqlQuery.filterQuery?.kuery?.expression ?? '',
+          language: selectedTimeline.kqlQuery.filterQuery?.kuery?.kind ?? 'kuery',
+        };
+    const newFilters = isEql ? [] : selectedTimeline.filters ?? [];
```

Comment:
This just helped me find a bug where we show `filter in` and `filter out` in the fields in EQL, but they don't actually do anything in that view... 😞